tensorflow/core/util/xla_config_proxy.h

```diff
@@ -0,0 +1,85 @@
+/* Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#ifndef TENSORFLOW_CORE_UTIL_XLA_CONFIG_PROXY_H
+#define TENSORFLOW_CORE_UTIL_XLA_CONFIG_PROXY_H
+
+#include <functional>
+#include "tensorflow/core/platform/default/logging.h"
+#include "tensorflow/core/platform/types.h"
+#include "tensorflow/core/protobuf/config.pb.h"
+
+namespace tensorflow {
+
+// A proxy class of XLA config.
+class XlaConfigProxy {
```

Comment:
I'm not sure I understand how you're planning to use this, do you mind elaborating a bit?

Here is what I had in mind though:  have a registry that lets XLA register an `std::function<OptimizerOptions::GlobalJitLevel(const GraphOptimizationPassOptions& options)>`.  Then Grappler can check if XLA is enabled using this registered `std::function`; if nothing is registered then it can assume that XLA is not enabled.

Reply:
Very similar. Just that we need a singleton (or at least some static functions) for registering this `std::function` callback. And I slightly generalize the registry class into `ConfigSetterRegistry`.

`ConfigSetterRegistry` is a singleton utility class to register a `std::function<bool(ConfigType&)>`; meaning that it allows each different config type to register a callback. In the current case, XLA registers a `std::function<bool(GlobalJitLevel&)>`, and TF can then invoke this `std::function` to update the GlobalJitLevel value.

Perhaps the generalization loses the readability. If that is the case, let me remove the `ConfigSetterRegistry` and simply make `XlaConfigProxy` a singleton. Please feel free to let me know what you think.

Reply:
I think this generality is not YAGNI for now.

Reply:
I see your point.